### PR TITLE
[BUGFIX][Server] Re-enable Transparent and Bgcolor parameter for GetMap

### PR DESCRIPTION
**Pull request by rldhont**
Fixed #15990 wms png transparency not honoured

And add tests for transparent and bgcolor in GetMap and GetLegendGraphic

### CODE REVIEW

**Comment by rldhont** [comment]
@dmarteau I have created this PR to save tests. Do you think we have to merge this first or after yours https://github.com/qgis/QGIS/pull/3866 ?

**Comment by dmarteau** [comment]
@rldhont Changes affecting wmsserver are beginning to be painful to port. I would prefer that changes on wms be postsponed until  #3866 is merged. So we can work with only one version of wms.

**Comment by rldhont** [comment]
@dmarteau I'm agree, it's much more to save bugfix and prepare tests.

File: src/server/qgswmsserver.cpp [hunk 1]
```diff
@@ -2139,6 +2140,37 @@ void QgsWmsServer::configureMapSettings( const QPaintDevice* paintDevice, QgsMap
...
   mapSettings.setExtent( mapExtent );
+
+  /* Define the background color
+   * Transparent or colored
+   */
+  //is format jpeg?
+  QString format = mParameters.value( QStringLiteral( "FORMAT" ) );
+  bool jpeg = format.compare( QLatin1String( "jpg" ), Qt::CaseInsensitive ) == 0
+              || format.compare( QLatin1String( "jpeg" ), Qt::CaseInsensitive ) == 0
+              || format.compare( QLatin1String( "image/jpeg" ), Qt::CaseInsensitive ) == 0;
```
Comment (sbrunner) [inline src/server/qgswmsserver.cpp]: The FORMAT parameter can relay be `jpg` or `jpeg`? Isn't invalid values?

Comment (rldhont) [inline src/server/qgswmsserver.cpp]: It's a copy of the old code...
We can clean it after merging the qgis server refactoring.

Comment (sbrunner) [inline src/server/qgswmsserver.cpp]: OK, +1 to remove after mergeing :-)

**Review by sbrunner — APPROVED** [review]

**Pull request merged**